### PR TITLE
[BUG][SIMS-#1181]Course load offering undefined Bug

### PR DESCRIPTION
**Pull request by guru-aot**
Course load offering undefined Bug

### CODE REVIEW

**Review by henryoforeh-aot — APPROVED** [review]

**Review by ann-aot — APPROVED** [review]

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts [hunk 1]
```diff
@@ -167,73 +167,79 @@ export class ECertFullTimeFileRecord extends ECertFileRecord {
...
-    record.repeatAppend(SPACE_FILLER, DATE_FORMAT.length); // Borrower Alternate Address Last Update, optional, not provided.
-    record.repeatAppend(SPACE_FILLER, 69); // Trailing space
-    return record.toString();
+      record.repeatAppend(SPACE_FILLER, DATE_FORMAT.length); // Borrower Address Last Update Date, optional, not provided.
+      record.repeatAppend(SPACE_FILLER, DATE_FORMAT.length); // Borrower Alternate Address Last Update, optional, not provided.
+      record.repeatAppend(SPACE_FILLER, 69); // Trailing space
+      return record.toString();
+    } catch (error: unknown) {
+      throw new Error(
+        `Error while creating record: ${this.documentNumber}. Error ${error}`,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts:241]: Can we change the message to explain what the number is? Something like:
```
`Error while creating record with document number: ${this.documentNumber}. Error ${error}`
```

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts [hunk 2]
```diff
@@ -167,73 +167,79 @@ export class ECertFullTimeFileRecord extends ECertFileRecord {
...
-    // Add the list of awards codes and values that always fill the same amount
-    // of slots defined on GRANT_AWARD_SLOTS. When there values available
-    for (let i = 0; i < AWARD_SLOTS; i++) {
-      const award = this.grantAwards.shift();
-      if (award) {
-        record.appendWithEndFiller(award.valueCode, 4, SPACE_FILLER);
-        record.appendWithStartFiller(award.valueAmount, 6, NUMBER_FILLER);
-      } else {
-        record.repeatAppend(SPACE_FILLER, 10); // Empty data for code(length=4)+amount(length=6) = 10 empty spaces.
+    try {
```
Comment (dheepak-aot) [inline sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts:170]: why did we add the try/catch? I didn't get it.

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts [hunk 3]
```diff
@@ -132,66 +132,72 @@ export class ECertPartTimeFileRecord extends ECertFileRecord {
...
-    record.repeatAppend(SPACE_FILLER, 5); // CSGP MP PDSE Amt, No longer used.
-    record.repeatAppend(SPACE_FILLER, 20); // Space Filler.
-    record.appendWithEndFiller(this.emailAddress, 75, SPACE_FILLER);
-    record.append("P"); // 'P' for part-time. Full time is done by another integration to another system.
-    record.appendDate(this.enrollmentConfirmationDate, DATE_FORMAT);
-    record.repeatAppend(SPACE_FILLER, 7); // EI Remittance Amt, optional, not provided.
-    record.appendWithStartFiller(this.courseLoad, 2, NUMBER_FILLER);
-    record.repeatAppend(SPACE_FILLER, 25); // Space Filler.
-    return record.toString();
+    try {
```
Comment (dheepak-aot) [inline sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts:135]: same here.

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍

File: sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts (hunk 2 above)
Comment (guru-aot) [reply sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-files/e-cert-file-record.ts:170]: to find the particular document number failing :)

**Review by dheepak-aot — APPROVED** [review]
LGTM 👍

**Pull request merged**